juju/juju
# Correct LXD server spec assignment

Fixes another place where we are using fluent options, but not assigning the result.

Included is a bunch of test reorganisation that was not actually necessary for this patch, but sets us up well for a forthcoming one.
- Mocks and some other test infra are moved into the `lxd` package from `lxd_test`.
- Mock generation is unified for the package and moved into `package_test.go`.
- Separate server test suites now exist for internal and external testing.

## QA steps

- Set up a remote LXD server (I did this on a MAAS-commissioned machine), ensuring access from the network, and setting a trust password.
- `juju add-cloud` and follow the prompts to use the new server.
- `juju add-credential <cloud-you-added-above>` follow the prompts for the interactive auth, entering the trust password.
- Check you can bootstrap to the cloud.

## Documentation changes

None

## Bug reference

N/A
